scripts/promise_batch_imports.py

```diff
@@ -52,8 +58,8 @@ def map_book_to_olbook(book, promise_id):
         **({'isbn_13': [isbn]} if is_isbn_13(isbn) else {}),
         **({'isbn_10': [book.get('ASIN')]} if asin_is_isbn_10 else {}),
         **({'title': title} if title else {}),
-        'authors': [{"name": book['ProductJSON'].get('Author') or '????'}],
-        'publishers': [book['ProductJSON'].get('Publisher') or '????'],
+        'authors': [{"name": clean_null(product_json.get('Author')) or '????'}],
+        'publishers': [clean_null(product_json.get('Publisher')) or '????'],
```

Comment:
Shouldn't this be `get('Publisher', '????')`  if `????` is supposed to be the fallback value?

I remember being worried that `????` might make it into live records, but I think some part of the code strips these out later. I can't remember if the `????` is useful before then though, would `None` do the same job here?

Reply:
The data can have things like `"Publisher": ""`. That would cause `.get('Publisher', '????')` to return `''` instead of `????`. That's why it's `or `'????'`.

Iirc, the `????` are handled down stream by the api; would be worth reinvestigating but considering out of scope of this issue.

Reply:
ok, that makes sense. The `????` makes things a little unclear what the intent is and threw me.